docs/pull_request_template.md

```diff
@@ -0,0 +1,3 @@
+## Checklist
```

Comment:
Since we don't have anything else in `docs/`, what do you think about putting this in `.github/` instead? I think some other GS repos use this location although I don't remember which.

Reply:
I just added `docs` in:

- https://github.com/giantswarm/template-operator
- https://github.com/giantswarm/template

I'd start with changing that here. I'm not attached to any of those locations.